### DISCUSSION TITLE
Service user calling `ListMyProjectOrgs` returns nothing

### DISCUSSION THREAD

**Discussion by startsUp** [zitadel/zitadel#9107] [Q&A]
I'm creating service users to authenticate CLI client for my application. The service user calls my backend to retrieve the list of orgs. My backend sends this request to Zitadel using `ListMyProjectOrgs`. Even though I've added given the service user the right authorizations (roles to my project) calling the `ListMyProjectOrgs` returns no results. On the other hand for human users that are logging in via the frontend and able to get the organization. What could be the problem?

```
res, err := o.zAuthClient.AuthService().ListMyProjectOrgs(zitadelClient.AuthorizedUserCtx(ctx.Request().Context()), &zAuth.ListMyProjectOrgsRequest{})
```

the CLI gets the token :

```
const (
	Scopes = "openid,email,profile,offline_access,urn:zitadel:iam:user:resourceowner,urn:zitadel:iam:org:project:id:zitadel:aud"
)

var scopeArr = strings.Split(Scopes, ",")

type ClientCredentials struct {
	ClientID     string
	ClientSecret string
	TokenURL     string
}

type Token string

// GetToken fetches an OAuth token using client credentials flow
func GetToken(ctx context.Context) (Token, error) {
	scopeArr = append(scopeArr, fmt.Sprintf("urn:zitadel:iam:org:project:id:%s:aud", viper.GetString(ZitadelProjectID)))
	config := clientcredentials.Config{
		ClientID:     viper.GetString(ClientID),
		ClientSecret: viper.GetString(ClientSecret),
		TokenURL:     fmt.Sprintf("%s/oauth/v2/token", viper.GetString(AuthURL)),
		Scopes:       scopeArr,
		AuthStyle:    oauth2.AuthStyleInHeader,
	}

	slog.Debug("Using oauth credentials config", "config", config)
	token, err := config.Token(ctx)
	if err != nil {
		return "", fmt.Errorf("failed to get token: %w", errors.WithStack(err))
	}

	slog.Debug("Retrieved token based on credentials config")
	return Token(token.AccessToken), nil
}
```

**Comment by hifabienne** [comment]
From what i remember this endpoint is about the project that the user is currently authenticated. So it takes the audience from the token and lists all the organizations the user has permission in that case.
In an oidc flow of a user the audience is always filled.
Depending on how the token is generated for the service user you might not have that information. you could request the audience with the audience scope. https://zitadel.com/docs/apis/openidoauth/scopes#reserved-scopes

**Comment by nasirovsh** [comment]
Were you able to solve this issue ? 
I am also trying to get Organization list with POST request to auth/v1/global/projectorgs/_search. With my own access token it works, but with Service User's PAT (Personal Access Token) it returns empty.

**Reply by hifabienne** [reply]
I am actually not sure if this works with a PAT.
If you want to get the roles we need to know for which project, but in a PAT you are not able to define the audience.
If you would use for example JWT Profile to authenticate your service user I think it should work

**Reply by VilleFTW** [reply]
> I am actually not sure if this works with a PAT. If you want to get the roles we need to know for which project, but in a PAT you are not able to define the audience. If you would use for example JWT Profile to authenticate your service user I think it should work

Hi Fabienne,

This seems to be more of a limitation. Is this planned to be changed with the upcoming release of v4 and v2 of Organization/Project/Instance Services? There are a couple more endpoints that could make use of a ServiceUser with a PAT being able to call them.
